base/constants_syncdocs.go

```diff
@@ -0,0 +1,85 @@
+	AttPrefix                        = SyncDocPrefix + "att:"                          // AttPrefix SG (v1) attachment data
+	Att2Prefix                       = SyncDocPrefix + "att2:"                         // Att2Prefix SG v2 attachment data
+	DCPBackfillSeqKey                = SyncDocPrefix + "dcp_backfill"                  // DCPBackfillSeqKey stores a BackfillSequences for a DCP feed
+	RepairBackupPrefix               = SyncDocPrefix + "repair:backup:"                // RepairBackupPrefix is the doc prefix used to store a backup of a repaired document
+	RepairDryRunPrefix               = SyncDocPrefix + "repair:dryrun:"                // RepairDryRunPrefix is the doc prefix used to store a repaired document in dry-run mode
+	SGRStatusPrefix                  = SyncDocPrefix + "sgrStatus:"                    // SGRStatusPrefix is the doc prefix used to store ISGR status documents
+)
+
+// Sync Gateway Metadata documents that should be GroupID scoped and accessed via the "WithGroupID" helper methods below
+const (
+	DCPCheckpointPrefixWithoutGroupID    = SyncDocPrefix + "dcp_ck:"   // DCPCheckpointPrefixWithoutGroupID stores a DCP checkpoint
+	SGCfgPrefixWithoutGroupID            = SyncDocPrefix + "cfg"       // SGCfgPrefixWithoutGroupID stores CfgSG/cbgt data
+	HeartbeaterPrefixWithoutGroupID      = SyncDocPrefix               // HeartbeaterPrefixWithoutGroupID stores a SG node heartbeat document
+	PersistentConfigPrefixWithoutGroupID = SyncDocPrefix + "dbconfig:" // PersistentConfigPrefixWithoutGroupID stores a database config
+	SyncFunctionKeyWithoutGroupID        = SyncDocPrefix + "syncdata"  // SyncFunctionKeyWithoutGroupID stores a copy of the Sync Function
+)
+
+// SyncFunctionKeyWithGroupID returns a doc ID to use when storing the sync function
+func SyncFunctionKeyWithGroupID(groupID string) string {
+	if groupID != "" {
+		return SyncFunctionKeyWithoutGroupID + ":" + groupID
+	}
+	return SyncFunctionKeyWithoutGroupID
+}
+
+// DCPCheckpointPrefixWithGroupID returns a doc ID prefix to use for DCP checkpoints
+func DCPCheckpointPrefixWithGroupID(groupID string) string {
+	if groupID != "" {
+		return DCPCheckpointPrefixWithoutGroupID + groupID + ":"
+	}
+	return DCPCheckpointPrefixWithoutGroupID
+}
+
+// SGCfgPrefixWithGroupID returns a doc ID prefix to use for CfgSG/cbgt documents
+func SGCfgPrefixWithGroupID(groupID string) string {
+	if groupID != "" {
+		return SGCfgPrefixWithoutGroupID + ":" + groupID + ":"
+	}
+	return SGCfgPrefixWithoutGroupID
+}
+
+// HeartbeaterPrefixWithGroupID returns a document prefix to use for heartbeat documents
+func HeartbeaterPrefixWithGroupID(groupID string) string {
+	if groupID != "" {
+		return HeartbeaterPrefixWithoutGroupID + groupID + ":"
+	}
+	return HeartbeaterPrefixWithoutGroupID
+}
+
+// PersistentConfigPrefixWithGroupID returns a document prefix to use to store database configs
+func PersistentConfigPrefixWithGroupID(groupID string) string {
```

Comment:
nit: in [rest/rest_tester_cluster_test.go](https://github.com/couchbase/sync_gateway/pull/5734/files#diff-09a205193ae1b1cf423fecffbca4b0ff45db506f179a47b283341d9da91196e8) and https://github.com/couchbase/sync_gateway/pull/5734/files#diff-75823786f84378e48731538d17b3378dda5061856fb67bbf591097e3a93ef5ec we use this as a key, not a prefix - so should it be `PersistentConfigKeyWithGroupID`?